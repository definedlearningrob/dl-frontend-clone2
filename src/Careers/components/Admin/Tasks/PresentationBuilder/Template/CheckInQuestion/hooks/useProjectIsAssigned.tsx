import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
