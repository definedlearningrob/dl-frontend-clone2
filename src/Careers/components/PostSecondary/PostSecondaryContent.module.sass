@import '@shared/stylesheets/config.sass'

$search-section-max-height: 200px
$section-max-height: 500px

.verticalList
  padding-right: $m-xxs !important

.content
  overflow: auto
  min-height: 100%
  max-height: 100%
  padding: $m-base
  display: grid
  gap: $m-base
  grid-template-columns: 1fr 340px
  grid-template-rows: min-content minmax(200px, 1fr)

  +media-breakpoint-up(xxxl)
    grid-template-columns: 2fr 1fr
    gap: $m-md
    padding: $m-md

  & .futureListSection,
  & .favoritesSection,
  & .applicationsSection
    background-color: $color-white
    padding: $m-base
    border-radius: $border-radius-sm

    +media-breakpoint-up(xxxl)
      padding: $m-md

.searchSection
  background-color: $color-white
  padding: $m-base
  border-radius: $border-radius-base

  +media-breakpoint-up(xxxl)
    padding: $m-md

.applicationsEnabled
  grid-template-rows: repeat(3, 1fr)

.futureListSection
  @extend .verticalList
  min-width: 0
  grid-row: span 2
  height: 100%

.favoritesSection
  min-width: 0
  padding-bottom: $m-sm !important

.applicationsSection
  grid-column: span 2
  &#{&}
    padding: 0 0 $m-xs 0 !important
