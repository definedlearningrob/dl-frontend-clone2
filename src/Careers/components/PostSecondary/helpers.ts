import { InstitutionApplication } from '@dc/graphql/student/queries/institutionApplications';
import { APPLICATIONS_TYPE, RECOMMENDER_TYPE } from '@dc/resources/enums';

export const checkIfHasEnoughTeachers = (application: InstitutionApplication) => {
  const teachers = application.recommenders.filter(
    (recommender) => recommender.type === RECOMMENDER_TYPE.TEACHER
  );

  return teachers.length >= (application.institution.minTeacherEval || 0);
};

export const checkIfCanBeMovedToCommonApp = (
  application: InstitutionApplication,
  hasAccountConnected: boolean
) =>
  application.type === APPLICATIONS_TYPE.DIRECT &&
  application.institution.commonAppEnabled &&
  hasAccountConnected;
