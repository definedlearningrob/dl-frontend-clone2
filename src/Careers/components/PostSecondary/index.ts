export * from './Search/Search';
export * from './FutureChecklist/FutureChecklist';
export * from './Applications/Applications';
export * from './Favorites/Favorites';
export * from './EmptyFutureChecklist/EmptyFutureChecklist';
export * from './EmptyFavorites/EmptyFavorites';
export * from './FavoritesList/FavoritesList';
export * from './FavoritesCarousel/FavoritesCarousel';
export * from '@dc/components/PostSecondary/ApplicationRecommenderList/ApplicationRecommenderList';
export * from './EmptyApplications/EmptyApplications';
export * from '@dc/components/PostSecondary/ApplicationDetailsModal/ApplicationDetailsModal';
export * from './PostSecondaryContent';
