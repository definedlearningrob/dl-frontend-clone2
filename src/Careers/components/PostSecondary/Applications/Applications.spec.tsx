import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { Applications } from '@dc/components/PostSecondary';
import { studentInfoMock } from '@dc/tests/mocks/studentMocks';
import { INSTITUTION_APPLICATIONS } from '@dc/graphql/student/queries/institutionApplications';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { INSTITUTION_APPLICATION_STATUS } from '@dc/resources/enums';
import { UPDATE_INSTITUTION_APPLICATION } from '@dc/graphql/student/mutations/updateInstitutionApplication';

const institutionApplicationsMockDirectToCommonApp = {
  request: {
    query: INSTITUTION_APPLICATIONS,
    variables: {
      page: 1,
      perPage: 100,
    },
  },
  result: {
    data: {
      institutionApplications: {
        nodes: [
          {
            id: '64',
            name: 'Harvard direct application to Common ',
            type: 'DIRECT',
            status: 'NOT_STARTED',
            institution: {
              commonAppApplicationUrl: 'https://test2.com',
              id: '444',
              minTeacherEval: 1,
              commonAppEnabled: true,
            },
            appliedAt: null,
            acceptsTeacherRecommendation: true,
            recommenders: [
              {
                email: 'alonzorockteach@cleverdemo.com',
                firstName: 'Alonzo',
                lastName: 'Rockteach',
                type: 'COUNSELOR',
              },
            ],
          },
        ],
      },
    },
  },
};

const institutionApplicationsMock = {
  request: {
    query: INSTITUTION_APPLICATIONS,
    variables: {
      page: 1,
      perPage: 100,
    },
  },
  result: {
    data: {
      institutionApplications: {
        nodes: [
          {
            id: '1',
            name: 'Texas Christian University',
            type: 'COMMON_APP',
            status: 'IN_PROGRESS',
            institution: {
              commonAppApplicationUrl: 'https://test.com',
              id: '111',
              minTeacherEval: 1,
              commonAppEnabled: true,
            },
            appliedAt: '2023-06-29T06:44:18Z',
            acceptsTeacherRecommendation: true,
            recommenders: [
              {
                email: 'alonzorockteach@cleverdemo.com',
                firstName: 'Alonzo',
                lastName: 'Rockteach',
                type: 'COUNSELOR',
              },
            ],
          },
          {
            id: '3',
            name: 'Yale University',
            type: 'COMMON_APP',
            status: 'IN_PROGRESS',
            institution: {
              commonAppApplicationUrl: 'https://test.com',
              id: '222',
              minTeacherEval: 2,
              commonAppEnabled: true,
            },
            appliedAt: '2023-07-06T07:40:13Z',
            acceptsTeacherRecommendation: true,
            recommenders: [
              {
                email: 'alonzorockteach@cleverdemo.com',
                firstName: 'Alonzo',
                lastName: 'Rockteach',
                type: 'COUNSELOR',
              },
            ],
          },
          {
            id: '63',
            name: 'Test direct application',
            type: 'DIRECT',
            status: 'NOT_STARTED',
            institution: {
              commonAppApplicationUrl: null,
              id: '333',
              minTeacherEval: 0,
              commonAppEnabled: false,
            },
            appliedAt: null,
            acceptsTeacherRecommendation: false,
            recommenders: [],
          },
          {
            id: '64',
            name: 'Harvard direct application to Common ',
            type: 'DIRECT',
            status: 'IN_PROGRESS',
            institution: {
              commonAppApplicationUrl: 'https://test2.com',
              id: '444',
              minTeacherEval: 1,
              commonAppEnabled: true,
            },
            appliedAt: null,
            acceptsTeacherRecommendation: true,
            recommenders: [
              {
                email: 'alonzorockteach@cleverdemo.com',
                firstName: 'Alonzo',
                lastName: 'Rockteach',
                type: 'COUNSELOR',
              },
            ],
          },
        ],
      },
    },
  },
};

const getDirectApplicationMock = (initialStatus: INSTITUTION_APPLICATION_STATUS) => ({
  request: {
    query: INSTITUTION_APPLICATIONS,
    variables: {
      username: undefined,
      page: 1,
      perPage: 100,
    },
  },
  result: {
    data: {
      institutionApplications: {
        nodes: [
          {
            id: '63',
            name: 'Test direct application',
            type: 'DIRECT',
            status: initialStatus,
            institution: {
              commonAppApplicationUrl: null,
              id: '1034',
              minTeacherEval: 0,
              commonAppEnabled: false,
              __typename: 'Institution',
            },
            appliedAt: null,
            acceptsTeacherRecommendation: false,
            recommenders: [],
            __typename: 'InstitutionApplication',
          },
        ],
      },
    },
  },
});

const updateInstitutionSpy = jest.fn();

const getUpdateInstitutionApplicationMock = (status: INSTITUTION_APPLICATION_STATUS) => ({
  request: {
    query: UPDATE_INSTITUTION_APPLICATION,
    variables: {
      input: { institutionApplicationId: '63', status: status },
    },
  },
  result() {
    updateInstitutionSpy();

    return {
      data: {
        updateInstitutionApplication: {
          institutionApplication: {
            status: status,
            name: 'Test direct application',
          },
        },
      },
    };
  },
});

const renderApplications = (mocks: MockedResponse[] = [], hasAccountConnected = true) =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={mocks}>
      <UserInfoProvider
        value={{
          userInfo: {
            ...studentInfoMock.result.data.userInfo,
            commonAppData: {
              ...studentInfoMock.result.data.userInfo.commonAppData,
              hasAccountConnected,
            },
          },
        }}>
        <Applications />
      </UserInfoProvider>
    </MockedProvider>
  );

describe('Applications', () => {
  afterEach(async () => {
    jest.clearAllMocks();
  });

  it('should render correctly', async () => {
    const { container } = renderApplications([studentInfoMock, institutionApplicationsMock]);

    const firstRow = await screen.findByText('Texas Christian University');

    expect(firstRow).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  describe('DIRECT applications', () => {
    it('should render opened dropdown menu correctly ', async () => {
      const { baseElement } = renderApplications([
        studentInfoMock,
        getDirectApplicationMock(INSTITUTION_APPLICATION_STATUS.NOT_STARTED),
      ]);

      const firstInstitution = await screen.findByText('Test direct application');

      expect(firstInstitution).toBeInTheDocument();

      const statusBadge = screen.getByRole('button', { name: 'Not Started' });

      //workaround for Radix Dropdown.Trigger
      userEvent.type(statusBadge, '{enter}');

      expect(screen.getByRole('button', { name: 'In Progress' })).toBeInTheDocument();
      expect(screen.getByRole('button', { name: 'Completed' })).toBeInTheDocument();

      expect(baseElement).toMatchSnapshot();
    });

    describe('should render dropdown menu with', function () {
      const testCases = [
        ['In Progress', 'Completed', INSTITUTION_APPLICATION_STATUS.NOT_STARTED, 'Not Started'],
        ['Not Started', 'Completed', INSTITUTION_APPLICATION_STATUS.IN_PROGRESS, 'In Progress'],
        ['Not Started', 'In Progress', INSTITUTION_APPLICATION_STATUS.SUBMITTED, 'Completed'],
      ];

      it.each(testCases)(
        '%s & %s items',
        async (firstExpectedBadge, secondExpectedBadge, initialStatus, initialBadge) => {
          renderApplications([
            studentInfoMock,
            getDirectApplicationMock(initialStatus as INSTITUTION_APPLICATION_STATUS),
          ]);

          const firstInstitution = await screen.findByText('Test direct application');

          expect(firstInstitution).toBeInTheDocument();

          const statusBadge = screen.getByRole('button', { name: initialBadge });

          // workaround for Radix Dropdown.Trigger
          await userEvent.type(statusBadge, '{arrowdown}');

          expect(screen.getByRole('button', { name: firstExpectedBadge })).toBeInTheDocument();
          expect(screen.getByRole('button', { name: secondExpectedBadge })).toBeInTheDocument();
        }
      );
    });

    describe('should change application status', () => {
      const testCases = [
        [
          'Not Started',
          'In Progress',
          INSTITUTION_APPLICATION_STATUS.NOT_STARTED,
          INSTITUTION_APPLICATION_STATUS.IN_PROGRESS,
        ],
        [
          'Not Started',
          'Completed',
          INSTITUTION_APPLICATION_STATUS.NOT_STARTED,
          INSTITUTION_APPLICATION_STATUS.SUBMITTED,
        ],
        [
          'In Progress',
          'Completed',
          INSTITUTION_APPLICATION_STATUS.IN_PROGRESS,
          INSTITUTION_APPLICATION_STATUS.SUBMITTED,
        ],
        [
          'In Progress',
          'Not Started',
          INSTITUTION_APPLICATION_STATUS.IN_PROGRESS,
          INSTITUTION_APPLICATION_STATUS.NOT_STARTED,
        ],
        [
          'Completed',
          'Not Started',
          INSTITUTION_APPLICATION_STATUS.SUBMITTED,
          INSTITUTION_APPLICATION_STATUS.NOT_STARTED,
        ],
        [
          'Completed',
          'In Progress',
          INSTITUTION_APPLICATION_STATUS.SUBMITTED,
          INSTITUTION_APPLICATION_STATUS.IN_PROGRESS,
        ],
      ];

      it.each(testCases)(
        'from %s to %s',
        async (initialBadge, expectedBadge, initialStatus, expectedStatus) => {
          renderApplications([
            studentInfoMock,
            getDirectApplicationMock(initialStatus as INSTITUTION_APPLICATION_STATUS),
            getUpdateInstitutionApplicationMock(expectedStatus as INSTITUTION_APPLICATION_STATUS),
          ]);

          const firstRow = await screen.findByText('Test direct application');

          expect(firstRow).toBeInTheDocument();

          const statusBadge = screen.getByRole('button', { name: initialBadge });

          // workaround for Radix Dropdown.Trigger
          userEvent.type(statusBadge, '{enter}');

          const dropdownMenu = screen.getByRole('menu', { name: initialBadge });

          expect(dropdownMenu).toBeInTheDocument();

          const statusToSet = screen.getByRole('button', { name: expectedBadge });

          expect(statusToSet).toBeInTheDocument();

          userEvent.click(statusToSet);

          await waitFor(() => {
            expect(updateInstitutionSpy).toHaveBeenCalledTimes(1);
          });
        }
      );
    });

    describe('should render action dropdown menu', () => {
      it('action context show on mouse enter', async () => {
        renderWithRouterAndReduxProvider(
          <MockedProvider mocks={[institutionApplicationsMockDirectToCommonApp]}>
            <UserInfoProvider
              value={{
                userInfo: {
                  ...studentInfoMock.result.data.userInfo,
                  commonAppData: {
                    ...studentInfoMock.result.data.userInfo.commonAppData,
                    hasAccountConnected: true,
                  },
                },
              }}>
              <Applications />
            </UserInfoProvider>
          </MockedProvider>
        );

        const firstRow = await screen.findByText('Harvard direct application to Common');

        userEvent.hover(firstRow);

        const contextMenuTrigger = await screen.findByTestId('application-context-menu');

        userEvent.type(contextMenuTrigger, '{enter}');

        const dropDownStatus = await screen.findByRole('link', { name: 'Change to Common App' });
        expect(dropDownStatus).toBeInTheDocument();

        userEvent.click(dropDownStatus);
        expect(window.location.pathname).toBe('/');
      });
    });
  });
});
