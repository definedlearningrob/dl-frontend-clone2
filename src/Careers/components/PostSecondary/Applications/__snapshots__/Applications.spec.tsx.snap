// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications DIRECT applications should render opened dropdown menu correctly  1`] = `
<body>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div>
    <div
      class="flex flex-col h-full"
    >
      <div
        class="flex justify-between mb-sm items-center px-base pt-base xxxl:px-md xxxl:pt-md"
      >
        <div
          class="flex flex-col gap-xs"
        >
          <h5
            class="heading"
          >
            Application Status
          </h5>
        </div>
      </div>
      <div
        class="applications"
      >
        <div
          class="shared-table"
        >
          <table
            class="shared-table__table !h-full"
          >
            <thead
              class="shared-table__head shared-table__head-cols-5"
            >
              <tr
                class="shared-table__row shared-table__row-0"
              >
                <th
                  class="shared-table__column -header-cell shared-table__column-1 tableHead"
                >
                  Future name
                </th>
                <th
                  class="shared-table__column -header-cell shared-table__column-2 tableHead"
                >
                  Date applied
                </th>
                <th
                  class="shared-table__column -header-cell shared-table__column-3 tableHead"
                >
                  Application Method
                </th>
                <th
                  class="shared-table__column -header-cell shared-table__column-4 tableHead"
                >
                  Status
                </th>
                <th
                  class="shared-table__column -header-cell shared-table__column-5 tableHead"
                />
              </tr>
            </thead>
            <tbody
              class="shared-table__body shared-table__body-cols-5"
            >
              <tr
                class="shared-table__row shared-table__row-1 tableRow"
                data-testid="list-item"
              >
                <td
                  class="shared-table__column shared-table__column-1 tableColumn"
                  data-testid="shared-table-column"
                >
                  <div>
                    <span
                      class="name"
                    >
                      Test direct application
                    </span>
                  </div>
                </td>
                <td
                  class="shared-table__column shared-table__column-2 tableColumn"
                  data-testid="shared-table-column"
                >
                  <span
                    data-state="closed"
                  >
                    <span>
                      N/A
                    </span>
                  </span>
                </td>
                <td
                  class="shared-table__column shared-table__column-3 tableColumn"
                  data-testid="shared-table-column"
                >
                  <span
                    class="applicationMethod"
                  >
                    Direct
                  </span>
                </td>
                <td
                  class="shared-table__column shared-table__column-4 tableColumn"
                  data-testid="shared-table-column"
                >
                  <div
                    class="statusColumn"
                  >
                    <div>
                      <button
                        aria-controls="radix-14"
                        aria-expanded="true"
                        aria-haspopup="menu"
                        class="flex items-center text-font-primary hover:text-primary-500 rounded-xs border border-solid border-transparent focus-visible:border-primary-500"
                        data-state="open"
                        id="radix-12"
                        type="button"
                      >
                        <div
                          class="font-medium leading-lg flex items-center flex items-center gap-xxs group-hover/statement:bg-white bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
                        >
                          Not Started
                        </div>
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </button>
                    </div>
                  </div>
                </td>
                <td
                  class="shared-table__column shared-table__column-5 tableColumn"
                  data-testid="shared-table-column"
                >
                  <div
                    class="flex gap-xs justify-end grow-1 w-full"
                  >
                    <div
                      class="invisible action"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="menu"
                        class="bg-white rounded-xs"
                        data-state="closed"
                        data-testid="application-context-menu"
                        id="radix-trigger"
                        type="button"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-sm w-sm"
                            />
                          </div>
                        </div>
                      </button>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content;"
  >
    <ul
      aria-labelledby="radix-12"
      aria-orientation="vertical"
      class="bg-white rounded-sm py-xxs border border-neutral-300 shadow-200 flex flex-col p-xxs"
      data-align="end"
      data-orientation="vertical"
      data-radix-menu-content=""
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-14"
      role="menu"
      style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); animation: none;"
      tabindex="-1"
    >
      <span
        class="font-bold text-xs py-xs px-sm"
      >
        Change to:
      </span>
      <li
        class="py-xs px-sm hover:bg-primary-200 focus-visible:!outline-none cursor-pointer rounded-xs border border-solid border-transparent focus-visible:border-primary-500 text-primary-500"
        data-highlighted=""
        data-orientation="vertical"
        data-radix-collection-item=""
        role="button"
        tabindex="0"
      >
        <div
          class="flex items-center gap-xxs text-xs leading-lg"
        >
          Not Started
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
        </div>
      </li>
      <li
        class="py-xs px-sm hover:bg-primary-200 focus-visible:!outline-none cursor-pointer rounded-xs border border-solid border-transparent focus-visible:border-primary-500"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="button"
        tabindex="-1"
      >
        <div
          class="flex items-center gap-xxs text-xs leading-lg"
        >
          In Progress
        </div>
      </li>
      <li
        class="py-xs px-sm hover:bg-primary-200 focus-visible:!outline-none cursor-pointer rounded-xs border border-solid border-transparent focus-visible:border-primary-500"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="button"
        tabindex="-1"
      >
        <div
          class="flex items-center gap-xxs text-xs leading-lg"
        >
          Completed
        </div>
      </li>
    </ul>
  </div>
  <span
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Applications should render correctly 1`] = `
<div>
  <div
    class="flex flex-col h-full"
  >
    <div
      class="flex justify-between mb-sm items-center px-base pt-base xxxl:px-md xxxl:pt-md"
    >
      <div
        class="flex flex-col gap-xs"
      >
        <h5
          class="heading"
        >
          Application Status
        </h5>
      </div>
    </div>
    <div
      class="applications"
    >
      <div
        class="shared-table"
      >
        <table
          class="shared-table__table !h-full"
        >
          <thead
            class="shared-table__head shared-table__head-cols-5"
          >
            <tr
              class="shared-table__row shared-table__row-0"
            >
              <th
                class="shared-table__column -header-cell shared-table__column-1 tableHead"
              >
                Future name
              </th>
              <th
                class="shared-table__column -header-cell shared-table__column-2 tableHead"
              >
                Date applied
              </th>
              <th
                class="shared-table__column -header-cell shared-table__column-3 tableHead"
              >
                Application Method
              </th>
              <th
                class="shared-table__column -header-cell shared-table__column-4 tableHead"
              >
                Status
              </th>
              <th
                class="shared-table__column -header-cell shared-table__column-5 tableHead"
              />
            </tr>
          </thead>
          <tbody
            class="shared-table__body shared-table__body-cols-5"
          >
            <tr
              class="shared-table__row shared-table__row-1 tableRow"
              data-testid="list-item"
            >
              <td
                class="shared-table__column shared-table__column-1 tableColumn"
                data-testid="shared-table-column"
              >
                <div>
                  <div
                    class="kicker sm secondary pendingLabel"
                  >
                    Needs attention
                  </div>
                  <span
                    class="name"
                  >
                    Texas Christian University
                  </span>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-2 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  data-state="closed"
                >
                  <span>
                    N/A
                  </span>
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-3 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  class="applicationMethod"
                >
                  <div />
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-4 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="statusColumn"
                >
                  <div
                    class="badgeWrapper"
                  >
                    <div
                      class="font-medium leading-lg flex items-center bg-secondary-200 text-secondary-500 rounded-sm p-xxs text-xs gap-xxs"
                    >
                      In Progress
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-5 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="flex gap-xs justify-end grow-1 w-full"
                >
                  <div
                    class="rounded-xs"
                  >
                    <span
                      data-state="closed"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 !bg-info-100 rounded-xs !text-info-500 p-xxs"
                          data-testid="icon"
                        >
                          <div
                            class="h-base w-base"
                          />
                        </div>
                      </div>
                    </span>
                  </div>
                  <div
                    class="invisible action"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="bg-white rounded-xs"
                      data-state="closed"
                      data-testid="application-context-menu"
                      id="radix-trigger"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 p-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class="shared-table__row shared-table__row-2 tableRow"
              data-testid="list-item"
            >
              <td
                class="shared-table__column shared-table__column-1 tableColumn"
                data-testid="shared-table-column"
              >
                <div>
                  <div
                    class="kicker sm secondary pendingLabel"
                  >
                    Needs attention
                  </div>
                  <span
                    class="name"
                  >
                    Yale University
                  </span>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-2 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  data-state="closed"
                >
                  <span>
                    N/A
                  </span>
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-3 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  class="applicationMethod"
                >
                  <div />
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-4 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="statusColumn"
                >
                  <div
                    class="badgeWrapper"
                  >
                    <div
                      class="font-medium leading-lg flex items-center bg-secondary-200 text-secondary-500 rounded-sm p-xxs text-xs gap-xxs"
                    >
                      In Progress
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-5 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="flex gap-xs justify-end grow-1 w-full"
                >
                  <div
                    class="rounded-xs"
                  >
                    <span
                      data-state="closed"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 !bg-info-100 rounded-xs !text-info-500 p-xxs"
                          data-testid="icon"
                        >
                          <div
                            class="h-base w-base"
                          />
                        </div>
                      </div>
                    </span>
                  </div>
                  <div
                    class="invisible action"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="bg-white rounded-xs"
                      data-state="closed"
                      data-testid="application-context-menu"
                      id="radix-trigger"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 p-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class="shared-table__row shared-table__row-3 tableRow"
              data-testid="list-item"
            >
              <td
                class="shared-table__column shared-table__column-1 tableColumn"
                data-testid="shared-table-column"
              >
                <div>
                  <span
                    class="name"
                  >
                    Test direct application
                  </span>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-2 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  data-state="closed"
                >
                  <span>
                    N/A
                  </span>
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-3 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  class="applicationMethod"
                >
                  Direct
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-4 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="statusColumn"
                >
                  <div>
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="flex items-center text-font-primary hover:text-primary-500 rounded-xs border border-solid border-transparent focus-visible:border-primary-500"
                      data-state="closed"
                      id="radix-2"
                      type="button"
                    >
                      <div
                        class="font-medium leading-lg flex items-center flex items-center gap-xxs group-hover/statement:bg-white bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
                      >
                        Not Started
                      </div>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-base w-base"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-5 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="flex gap-xs justify-end grow-1 w-full"
                >
                  <div
                    class="invisible action"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="bg-white rounded-xs"
                      data-state="closed"
                      data-testid="application-context-menu"
                      id="radix-trigger"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 p-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class="shared-table__row shared-table__row-4 tableRow"
              data-testid="list-item"
            >
              <td
                class="shared-table__column shared-table__column-1 tableColumn"
                data-testid="shared-table-column"
              >
                <div>
                  <div
                    class="kicker sm secondary pendingLabel"
                  >
                    Needs attention
                  </div>
                  <span
                    class="name"
                  >
                    Harvard direct application to Common 
                  </span>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-2 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  data-state="closed"
                >
                  <span>
                    N/A
                  </span>
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-3 tableColumn"
                data-testid="shared-table-column"
              >
                <span
                  class="applicationMethod"
                >
                  Direct
                </span>
              </td>
              <td
                class="shared-table__column shared-table__column-4 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="statusColumn"
                >
                  <div>
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="flex items-center text-font-primary hover:text-primary-500 rounded-xs border border-solid border-transparent focus-visible:border-primary-500"
                      data-state="closed"
                      id="radix-4"
                      type="button"
                    >
                      <div
                        class="font-medium leading-lg flex items-center flex items-center gap-xxs group-hover/statement:bg-white bg-secondary-200 text-secondary-500 rounded-sm p-xxs text-xs gap-xxs"
                      >
                        In Progress
                      </div>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-base w-base"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
              <td
                class="shared-table__column shared-table__column-5 tableColumn"
                data-testid="shared-table-column"
              >
                <div
                  class="flex gap-xs justify-end grow-1 w-full"
                >
                  <div
                    class="rounded-xs"
                  >
                    <span
                      data-state="closed"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 !bg-info-100 rounded-xs !text-info-500 p-xxs"
                          data-testid="icon"
                        >
                          <div
                            class="h-base w-base"
                          />
                        </div>
                      </div>
                    </span>
                  </div>
                  <div
                    class="invisible action"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="bg-white rounded-xs"
                      data-state="closed"
                      data-testid="application-context-menu"
                      id="radix-trigger"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 p-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
